### PR TITLE
IOS - Allow local files from static server.

### PR DESCRIPTION
**Pull request by tbatch92**
Local files might be located on a static server (eg using https://www.npmjs.com/package/react-native-static-server). In these cases the file will not be prefixed with file:// but are still playable with a network connection.

### CODE REVIEW

**Comment by Guichaguri** [comment]
Does iOS have a restriction to localhost URLs? I thought passing the URL directly would work as expected

**Comment by tbatch92** [comment]
A use case of a recent project was playing downloaded HLS tracks. 

The problem we ran into was if completely offline (no wifi or mobile data) the track would not be played at all if you were "adding". However if the track was added whilst online, it would still play after the connection was turned off. 

Seems to be attributed to [line 95 here](https://github.com/react-native-kit/react-native-track-player/blob/dev/ios/RNTrackPlayer/Vendor/AudioPlayer/player/AudioPlayer.swift), which checks if you're online or if the URL is flagged as 'isLocal'.

File: ios/RNTrackPlayer/Models/MediaURL.swift [hunk 1]
```diff
@@ -12,19 +12,19 @@ struct MediaURL {
...
+
         if let localObject = object as? [String: Any] {
             isLocal = true
             let uri = localObject["uri"] as! String
             let encodedURI = uri.addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)!
             value = URL(string: encodedURI)!
         } else {
             let url = object as! String
-            isLocal = url.lowercased().hasPrefix("file://")
+            isLocal = url.lowercased().hasPrefix("file://") || url.lowercased().hasPrefix("http://localhost")
```
Comment (dcvz) [inline ios/RNTrackPlayer/Models/MediaURL.swift]: What if someone was to use ‘https://localhost’?

Comment (Guichaguri) [reply ios/RNTrackPlayer/Models/MediaURL.swift]: And how about `http://192.168.0.1` or `https://192.168.0.1`?

**Comment by bradleyflood** [comment]
Closing this as outdated. Thanks all.

**Pull request closed without merging**